File: frontend/src/types/item.ts
import type { ApiResponse, PaginatedResponse, PaginationParams } from "./api";

/** Mirror backend/app/schemas/item.py */

export type ItemStatus = "draft" | "active" | "archived";

export interface ItemCreate {
  title: string;
  description?: string | null;
  status?: ItemStatus;
}

export interface ItemUpdate {
  title?: string;
  description?: string | null;
  status?: ItemStatus;
}

export interface ItemRead {
  id: string;
  title: string;
  description: string | null;
  status: ItemStatus;
  owner_id: string;
  created_at: string;
  updated_at: string;
}

export interface ItemListParams extends PaginationParams {
  status?: ItemStatus;
}

export type ItemResponse = ApiResponse<ItemRead>;
export type ItemListResponse = PaginatedResponse<ItemRead>;
export type ItemDeleteResponse = ApiResponse<{ deleted: boolean }>;

File: backend/app/schemas/item.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, Field

from app.schemas.common import PaginationParams

ItemStatus = Literal["draft", "active", "archived"]


class ItemCreate(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=5000)
    status: ItemStatus = "draft"


class ItemUpdate(BaseModel):
    title: str | None = Field(default=None, min_length=1, max_length=200)
    description: str | None = Field(default=None, max_length=5000)
    status: ItemStatus | None = None


class ItemRead(BaseModel):
    id: str
    title: str
    description: str | None
    status: ItemStatus
    owner_id: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ItemListParams(PaginationParams):
    status: ItemStatus | None = None
